### PR TITLE
Various Add-ons: Tweak previously added toString() in logger usage

### PR DESCRIPTION
**Pull request by kingthorin**
Related to zaproxy/zaproxy#6376

I believe all impacted add-ons (with the exception of pascanrulesAlpha) had either a logging or maintenance changes not already (in the unreleased section).

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by thc202** [comment]
The `pscanrulesAlpha` add-on is being updated in #2774 so not worth doing changes here?

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/RelativePathConfusionScanRule.java [hunk 1]
```diff
@@ -207,7 +207,7 @@ public void scan() {
         log.debug(
                 "Checking [{}] [{}], for Relative Path Confusion issues",
                 originalMsg.getRequestHeader().getMethod(),
-                originalMsg.getRequestHeader().getURI().toString());
+                originalMsg.getRequestHeader());
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/RelativePathConfusionScanRule.java:210]: `.getURI()`

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/BackupFileDisclosureScanRule.java [hunk 2]
```diff
@@ -800,7 +800,7 @@ private void findBackupFile(HttpMessage originalMessage) throws Exception {
         } catch (Exception e) {
             log.error(
                     "Some error occurred when looking for a backup file for '{}'",
-                    originalMessage.getRequestHeader().getURI().toString(),
+                    originalMessage.getRequestHeader(),
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/BackupFileDisclosureScanRule.java:803]: `.getURI()`

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/RelativePathConfusionScanRule.java (hunk 1 above)
Comment (thc202) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/RelativePathConfusionScanRule.java:210]: In this class there's another log statement that could be tweaked:
```Java
log.debug("The file extension of {} is {}", baseUri.getURI().toString(), fileext);
```
`baseUri.getURI().toString()` → `baseUri`

**Comment by kingthorin** [comment]
Reverted pscanrulesAlpha changes and fixed the few things you noted.

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/RelativePathConfusionScanRule.java [hunk 3]
```diff
@@ -223,7 +223,7 @@ public void scan() {
             // Confusion
             // (based on the instances of this that I've in seen in the wild, at least)
             if (fileext != null && fileext.length() > 0) {
-                log.debug("The file extension of {} is {}", baseUri.getURI().toString(), fileext);
+                log.debug("The file extension of {} is {}", baseUri.getURI(), fileext);
```
Comment (thc202) [inline addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/RelativePathConfusionScanRule.java:226]: The `getURI()` can be removed too.

**Comment by kingthorin** [comment]
Done (hopefully 😀 )

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**